website/dynamo.py

```diff
@@ -1113,18 +1216,29 @@ def validate_only_sort_key(conds, sort_key):
         raise RuntimeError(f"Conditions only allowed on sort key {sort_key}, got: {list(conds)}")
 
 
-def encode_page_token(x):
-    """Encode a compound key page token (dict) to a string."""
+def encode_page_token(x, inverted):
+    """Encode a compound key page token (dict) to a string.
+
+    'inverted' indicates whether the page token goes in the other direction
+    than the normally accepted query direction. If the query normally goes
+    forward, a query with an inverted page token would go backwards; but if the
+    query normally goes backwards (reverse=True), a query with an inverted page token
+    would go forward.
+    """
     if x is None:
         return None
-    return base64.urlsafe_b64encode(json.dumps(x).encode("utf-8")).decode("ascii")
+    return ('-' if inverted else '') + base64.urlsafe_b64encode(json.dumps(x).encode("utf-8")).decode("ascii")
```

Comment:
This is just a trick for us to know in which direction we wanna retrieve results right?

Reply:
Yep!